red4ext-sys/src/interop.rs

```diff
@@ -46,6 +48,97 @@ unsafe impl ExternType for CName {
     type Kind = cxx::kind::Trivial;
 }
 
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+#[repr(C)]
+pub struct ResRef(RaRef);
+
+impl ResRef {
+    pub fn new(path: &str) -> Result<Self, ResourcePathError> {
+        Ok(Self(RaRef::new(path)?))
+    }
+}
+
+unsafe impl ExternType for ResRef {
+    type Id = type_id!("RED4ext::ResRef");
+    type Kind = cxx::kind::Trivial;
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+#[repr(C)]
+pub struct RaRef(ResourcePath);
+
+impl RaRef {
+    fn new(path: &str) -> Result<Self, ResourcePathError> {
+        Ok(Self(ResourcePath::try_new(path)?))
+    }
+}
+
+#[derive(Debug, Clone, Default, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct ResourcePath {
+    hash: u64,
+}
+
+impl ResourcePath {
+    const MAX_LENGTH: usize = 216;
+
+    /// accepts non-sanitized path of any length,
+    /// but final sanitized path length must be equals or inferior to 216 bytes
+    fn try_new(path: &str) -> Result<Self, ResourcePathError> {
```

Comment:
I'd just name it `new`,  adding a try_ prefix seems redundant and I don't think is a common practice

Reply:
The way the name try_new is used in standard library is to distinguish a Result constructor from one that panics, but here we only have a single constructor so there is no need to have a distinguished name. If you look at failing constructors in the std, you'll see that they never have a try prefix (File::open, OpenOptions::open). Using a try prefix is also never used on other methods that return Result like `fmt`, unless the name is already taken like `from`.
So, I'd avoid using a `try_` prefix unless it's used to distinguish a panicking operation from one that returns Result to be in line with standard library and major crates

Reply:
you're absolutely right, addressed in latest commits.